### ISSUE TITLE
OLAP mode still enforces TX timeouts (-queryserver-config-query-timeout)

### ISSUE THREAD

**Issue by maxenglander** [vitessio/vitess#10945]
(Clone of internal PlanetScale issue.)

Background:

  * For a long time, you could not use `workload=olap` with transactions
  * This changed for Vitess 13.0, see https://github.com/vitessio/vitess/pull/9115

Issue:
  * Since `workload=olap` bypasses the query timeouts (`-queryserver-config-query-timeout`) and also row limits, the natural assumption is that it also bypasses the transaction timeout.
  * This is not the case, e.g. for a tablet where the `-queryserver-config-transaction-timeout` is `10`:
```
mysql> set workload=olap;                   
Query OK, 0 rows affected (0.00 sec)

mysql> begin ; select * from t1; select sleep(35); commit;
Query OK, 0 rows affected (0.00 sec)

+----+------+
| c1 | c2   |
+----+------+
|  3 |    4 |
|  4 |    5 |
|  5 |    6 |
|  6 |    7 |
+----+------+
4 rows in set (0.00 sec)

ERROR 1317 (70100): vttablet: rpc error: code = Canceled desc = (errno 2013) due to context deadline exceeded, elapsed time: 10.000470709s, killing query ID 5463 (CallerID: user)
ERROR 1317 (70100): vttablet: rpc error: code = Aborted desc = transaction 1657804966728853781: ended at 2022-07-14 12:42:04.175 PDT (unlocked closed connection) (CallerID: user)
```
  * Workaround is to have OLAP mode clients not use transactions at all, this could be reasonable in some cases.
  * However, many of the use-cases for OLAP transactions are "streaming-transactional", e.g. Spark or Sqoop both use transactions when streaming its rows out of tables.  It is more challenging for average users to modify Spark or Sqoop to not use transactions.

Note that PSDB is also affected, since we set the `-queryserver-config-transaction-timeout` relatively low (20 secs), vs a high query timeout.

TL;DR:
  * There should be a way of having longer (or infinite?) transaction timeouts in OLAP mode without having to raise `-queryserver-config-transaction-timeout`, which would affect OLTP mode clients adversely.
  * We should also update the documentation accordingly.  Right now it just says:
```
By default, Vitess sets some intentional restrictions on the execution time and number of rows that a query can return. This default workload mode is called OLTP. This can be disabled by setting the workload to OLAP
```

**Comment by deepthi** [comment]
Fixed in #10946
